app/src/main/java/com/jerboa/ui/components/registrationapplications/RegistrationApplications.kt

```diff
@@ -0,0 +1,496 @@
+                        horizontalArrangement = spacedBy(MEDIUM_PADDING),
+                    ) {
+                        Text(
+                            text = stringResource(R.string.deny_reason) + ": ",
+                            style = MaterialTheme.typography.labelLarge,
+                        )
+
+                        MyMarkdownText(
+                            markdown = reason,
+                            onClick = {},
+                        )
+                    }
+                }
+            }
+        }
+
+        if (showDenyReasonField) {
+            MarkdownTextField(
+                text = denyReason,
+                onTextChange = { denyReason = it },
+                account = account,
+                modifier = Modifier.fillMaxWidth(),
+                placeholder = stringResource(R.string.type_your_reason),
+            )
+        }
+
+        Row(
+            horizontalArrangement = spacedBy(MEDIUM_PADDING),
+        ) {
+            val showApproveButton = app.admin_id == null || !accepted
+            val showDenyButton = app.admin_id == null || accepted
+
+            if (showApproveButton) {
+                OutlinedButton(
+                    onClick = {
+                        onApproveClick(
+                            ApproveRegistrationApplication(
+                                id = app.id,
+                                approve = true,
+                            ),
+                        )
+                    },
+                ) {
+                    Text(stringResource(R.string.approve))
+                }
+            }
+
+            if (showDenyButton) {
+                val errButtonColor = MaterialTheme.colorScheme.error
+                OutlinedButton(
+                    onClick = {
+                        // If the reason isn't shown first, show it
+                        if (!showDenyReasonField) {
+                            showDenyReasonField = true
+                        }
+                        // If it is shown, then send the form
+                        else {
+                            showDenyReasonField = false
+                            onApproveClick(
+                                ApproveRegistrationApplication(
+                                    id = app.id,
+                                    approve = false,
+                                    deny_reason = denyReason.text,
+                                ),
+                            )
+                        }
+                    },
+                    colors = ButtonDefaults.outlinedButtonColors(contentColor = errButtonColor),
+                    border = BorderStroke(1.dp, errButtonColor),
+                ) {
+                    Text(stringResource(R.string.deny))
+                }
+            }
+        }
+    }
+    HorizontalDivider(modifier = Modifier.padding(bottom = SMALL_PADDING))
+}
+
+@Preview
```

Comment:
I set these up, with a bunch of sample data too, but unfortunately android studio can't render them (I think because of the markdown view)